common/lib/xmodule/xmodule/js/src/video/08_video_auto_advance_control.js

```diff
@@ -0,0 +1,111 @@
+(function(requirejs, require, define) {
+    'use strict';
+    define(
+    'video/08_video_auto_advance_control.js', [],
+function() {
+    /**
+     * Auto advance control module.
+     * @exports video/08_video_auto_advance_control.js
+     * @constructor
+     * @param {object} state The object containing the state of the video player.
+     * @return {jquery Promise}
+     */
+    var AutoAdvanceControl = function(state) {
+        if (!(this instanceof AutoAdvanceControl)) {
+            return new AutoAdvanceControl(state);
+        }
+
+        _.bindAll(this, 'onClick', 'destroy', 'autoPlay', 'autoAdvance');
+        this.state = state;
+        this.state.videoAutoAdvanceControl = this;
+        this.initialize();
+
+        return $.Deferred().resolve().promise();
+    };
+
+    AutoAdvanceControl.prototype = {
+        template: [
+            '<button class="control auto-advance" aria-disabled="false" title="',
+            gettext('Auto-advance'),
+            '">',
+            '<span class="label" aria-hidden="true">', gettext('Auto-advance'), '</span>',
+            '</button>'
+        ].join(''),
```

Comment:
Please use the UI toolkit to interpolate this template and ensure proper escaping:

```
        template: HtmlUtils.interpolateHtml(
            HtmlUtils.HTML([
                '<button class="control auto-advance" aria-disabled="false" title="',
                '{autoAdvanceText}',
                '">',
                '<span class="label" aria-hidden="true">',
                '{autoAdvanceText}',
                '</span>',
                '</button>'].join('')),
            {
                autoAdvanceText: gettext('Auto-advance')
            }
        ),
```